src/protocol-parser/protocol-parser.ts

```diff
@@ -52,1048 +45,244 @@ export function parseObject<T extends ZodType>(
-    type: zod.literal('pointer'),
-    id: zod.string(),
-    parameters: PointerParametersSchema.optional(),
-    actions: zod.array(PointerSourceActionSchema),
-  });
-
-  // input.KeySourceAction = (
-  //   input.PauseAction //
-  //   input.KeyDownAction //
-  //   input.KeyUpAction
-  // )
-  const KeySourceActionSchema = zod.discriminatedUnion('type', [
-    PauseActionSchema,
-    KeyDownActionSchema,
-    KeyUpActionSchema,
-  ]);
-
-  // input.KeySourceActions = {
-  //   type: "key",
-  //   id: text,
-  //   actions: [*input.KeySourceAction]
-  // }
-  const KeySourceActionsSchema = zod.object({
-    type: zod.literal('key'),
-    id: zod.string(),
-    actions: zod.array(KeySourceActionSchema),
-  });
+  export function parsePerformActionsParams(params: unknown) {
+    return parseObject(
+      params,
+      WebDriverBidi.Input.PerformActionsParametersSchema
+    ) as Protocol.Input.PerformActionsParameters;
+  }
 
-  // input.NoneSourceAction = input.PauseAction
-  const NoneSourceActionSchema = PauseActionSchema;
+  export function parseReleaseActionsParams(
+    params: unknown
+  ): Protocol.Input.ReleaseActionsParameters {
+    return parseObject(
+      params,
+      WebDriverBidi.Input.ReleaseActionsParametersSchema
+    );
+  }
+}
 
-  // input.NoneSourceActions = {
-  //   type: "none",
-  //   id: text,
-  //   actions: [*input.NoneSourceAction]
-  // }
-  const NoneSourceActionsSchema = zod.object({
-    type: zod.literal('none'),
-    id: zod.string(),
-    actions: zod.array(NoneSourceActionSchema),
+export namespace Cdp {
+  const SendCommandRequestSchema = zod.object({
+    // Allowing any cdpMethod, and casting to proper type later on.
+    method: zod.string(),
+    // `passthrough` allows object to have any fields.
+    // https://github.com/colinhacks/zod#passthrough
+    params: zod.object({}).passthrough().optional(),
+    session: zod.string().optional(),
   });
 
-  // input.SourceActions = (
-  //   input.NoneSourceActions //
-  //   input.KeySourceActions //
-  //   input.PointerSourceActions //
-  //   input.WheelSourceActions
-  // )
-  const SourceActionsSchema = zod.discriminatedUnion('type', [
-    NoneSourceActionsSchema,
-    KeySourceActionsSchema,
-    PointerSourceActionsSchema,
-    WheelSourceActionsSchema,
-  ]);
-
-  // input.PerformActionsParameters = {
-  //   context: browsingContext.BrowsingContext,
-  //   actions: [*input.SourceActions]
-  // }
-  const PerformActionsParametersSchema = zod.object({
-    context: CommonDataTypes.BrowsingContextSchema,
-    actions: zod.array(SourceActionsSchema),
+  const GetSessionRequestSchema = zod.object({
+    context: WebDriverBidi.BrowsingContext.BrowsingContextSchema,
   });
 
-  export function parsePerformActionsParams(
-    params: unknown
-  ): InputTypes.PerformActionsParameters {
-    return parseObject(params, PerformActionsParametersSchema);
+  export function parseSendCommandRequest(params: unknown) {
```

Comment:
Does the compiler complain if you explicitly type annotate the return and remove the cast below?

(This comment applies to all `parseObject`.)

Reply:
Yea, because the an enum values doesn't equate to its string value; e.g. `Root` may equal `root`, but `'root' != Root`